librocksdb_sys/crocksdb/c.cc

```diff
@@ -3779,4 +3798,109 @@ int crocksdb_keyversions_type(const crocksdb_keyversions_t *kvs, int index) {
   return kvs->rep[index].type;
 }
 
+struct SstFileSeqNoModifier {
+  SstFileSeqNoModifier(Env *env, ColumnFamilyData *cfd):env_(env), cfd_(cfd) { }
+
+  Status GetTableReader(std::string file, std::unique_ptr<TableReader> *table_reader) {
+    // Get External Sst File Size
+    uint64_t file_size;
+    auto status = env_->GetFileSize(file, &file_size);
+    if (!status.ok()) {
+      return status;
+    }
+
+    // Open External Sst File
+    std::unique_ptr<RandomAccessFile> sst_file;
+    std::unique_ptr<RandomAccessFileReader> sst_file_reader;
+    status = env_->NewRandomAccessFile(file, &sst_file, env_options_);
+    if (!status.ok()) {
+      return status;
+    }
+    sst_file_reader.reset(new RandomAccessFileReader(std::move(sst_file), file));
+
+    // Get Table Reader
+    status = cfd_->ioptions()->table_factory->NewTableReader(
+        TableReaderOptions(*cfd_->ioptions(), env_options_,
+                           cfd_->internal_comparator()),
+        std::move(sst_file_reader), file_size, table_reader);
+    return status;
+  }
+
+  Status FindSeqNoInfo(TableReader *table_reader, uint64_t *ori_seq_no, uint64_t *offset) {
+    // Get the external file properties
+    auto props = table_reader->GetTableProperties();
+    const auto& uprops = props->user_collected_properties;
+    // Validate version and seqno offset
+    auto version_iter = uprops.find(ExternalSstFilePropertyNames::kVersion);
+    if (version_iter == uprops.end()) {
+      return Status::Corruption("External file version not found");
+    }
+    uint32_t version = DecodeFixed32(version_iter->second.c_str());
+    if (version != 2) {
+      return Status::NotSupported("External file version should be 2");
+    }
+
+    auto seqno_iter = uprops.find(ExternalSstFilePropertyNames::kGlobalSeqno);
+    *ori_seq_no = DecodeFixed64(seqno_iter->second.c_str());
+    *offset = props->properties_offsets.at(ExternalSstFilePropertyNames::kGlobalSeqno);
+    if (*offset == 0) {
+      return Status::Corruption("Was not able to find file global seqno field");
+    }
+    return Status::OK();
+  }
+
+  Status ModifySeqNo(std::string file, uint64_t ori_seq_no, uint64_t offset, uint64_t seq_no) {
```

Comment:
Don't expose too many details, from a user's perspective, it should be straightforward like:

```
ExternalSstFile file(...);
file.Open(filename);
file.UpdateSeqno(seqno);
```